.github/workflows/push-to-master.yml

```diff
@@ -1,39 +1,49 @@
 name: push-to-master
-on:
-  push:
-    branches:
-      - master
+on: pull_request
 jobs:
   test-ray-master:
     runs-on: ubuntu-latest
+    defaults:
+      run:
+        shell: bash -l {0}
     steps:
       - uses: actions/checkout@v2
         with:
           fetch-depth: 1
-      - shell: bash -l {0}
-        run: docker build -f .github/workflows/ray-master.dockerfile -t ray_docker_image:nightly .
-      - uses: addnab/docker-run-action@v2
+      - uses: conda-incubator/setup-miniconda@v2
         with:
-          image: ray_docker_image:nightly
-          run: |
-            pytest modin/experimental/xgboost/test/test_default.py
-            pytest modin/pandas/test/dataframe/test_binary.py
-            pytest modin/pandas/test/dataframe/test_default.py
-            pytest modin/pandas/test/dataframe/test_indexing.py
-            pytest modin/pandas/test/dataframe/test_iter.py
-            pytest modin/pandas/test/dataframe/test_join_sort.py
-            pytest modin/pandas/test/dataframe/test_map_metadata.py
-            pytest modin/pandas/test/dataframe/test_reduction.py
-            pytest modin/pandas/test/dataframe/test_udf.py
-            pytest modin/pandas/test/dataframe/test_window.py
-            pytest modin/pandas/test/test_series.py
-            pytest modin/pandas/test/test_rolling.py
-            pytest modin/pandas/test/test_concat.py
-            pytest modin/pandas/test/test_groupby.py
-            pytest modin/pandas/test/test_reshape.py
-            pytest modin/pandas/test/test_general.py
-            pytest modin/pandas/test/test_io.py
-            pytest modin/experimental/pandas/test/test_io_exp.py
+          activate-environment: modin
+          environment-file: requirements/requirements-no-engine.yml
+          python-version: 3.7
+          channel-priority: strict
+          use-only-tar-bz2: true # IMPORTANT: This needs to be set for caching to work properly!
+      - name: install Ray nightly build
+        run: pip install https://s3-us-west-2.amazonaws.com/ray-wheels/latest/ray-2.0.0.dev0-cp37-cp37m-manylinux2014_x86_64.whl
```

Comment:
What is the reason for the change from docker to `pip install`? Also this tests 2.0.0 and would have to be updated when they release it...

Reply:
pip installable ray version was used because of problems with creating environment inside Ray docker container (shortly described in the PR header). Yes, after new Ray release we will need to correct this line, but we already have hardcoded `pip install` command in the `test-ray-client` CI job.